src/commands/router.ts

```diff
@@ -88,7 +88,15 @@ export async function onInteractionReceived(interaction: Interaction) {
 
     await interaction.deferReply();
 
-    if (targetCommand.requiresAdmin && !isAdmin(interaction.member as GuildMember)) {
+    const subCommandRequiresAdmin =
```

Comment:
Could this be written into a function in `utils/commands`?

I haven't tested this, but something like

```typescript
function requiresAdminPermissions(command: BaseCommand, subCommandName: string | null) {
  if (!(command instanceof AggregateCommand)) {
    return !!command.requiresAdmin;
  }

  return (
    !!subCommandName &&
    !!command.subCommands.find(s => s.slashCommand.name === subCommandName)?.requiresAdmin
  );
}

```